.github/workflows/ci.yml

```diff
@@ -0,0 +1,74 @@
+# Find full documentation here https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
+name: CI
+
+on:
+    pull_request:
+
+    # Manual invocation.
+    workflow_dispatch:
+
+    push:
+        branches:
+            - main
+jobs:
+    CI:
+        runs-on: ubuntu-latest
+        permissions:
+            # required by aws-actions/configure-aws-credentials
+            id-token: write
+            contents: read
+        steps:
+            - uses: actions/checkout@v3
+
+            - uses: aws-actions/configure-aws-credentials@v2
```

Comment:
Also a [v4](https://github.com/aws-actions/configure-aws-credentials/releases) of this.